### PR TITLE
Feature/redis command log

### PR DESCRIPTION
**Pull request by jaquadro**

### CODE REVIEW

File: lib/stores/redisWorkerStore.js [hunk 1]
```diff
@@ -67,18 +69,27 @@ RedisWorkerStore.prototype.readItems = function (key, subkey, partialMatch) {
   }
 
   function zrangebylex(key, min, max) {
-    self.logCommand('readItems', ['zrangebylex', key, min, max]);
-    return self.redis.zrangebylexAsync(key, min, max);
+    const entry = self.logCommand('readItems', ['zrangebylex', key, min, max]);
+    return self.redis.zrangebylexAsync(key, min, max).then((result) => {
```
Comment (gotrevor) [inline lib/stores/redisWorkerStore.js]: `tap`?

File: lib/stores/redisWorkerStore.js [hunk 2]
```diff
@@ -67,18 +69,27 @@ RedisWorkerStore.prototype.readItems = function (key, subkey, partialMatch) {
...
+      entry.result = _.clone(result);
```
Comment (gotrevor) [inline lib/stores/redisWorkerStore.js]: This seems a little dodgy.  Mutating the return from the logCommand, in order to records the return value.

I won't hold the PR up for this, but do you agree it seems a bit odd?

File: lib/stores/redisWorkerStore.js [hunk 3]
```diff
@@ -67,18 +69,27 @@ RedisWorkerStore.prototype.readItems = function (key, subkey, partialMatch) {
   }
 
   function zrangebylex(key, min, max) {
-    self.logCommand('readItems', ['zrangebylex', key, min, max]);
-    return self.redis.zrangebylexAsync(key, min, max);
+    const entry = self.logCommand('readItems', ['zrangebylex', key, min, max]);
```
Comment (gotrevor) [inline lib/stores/redisWorkerStore.js]: s/entry/logEntry/   <-- would be every so slightly clearer.

File: lib/stores/redisWorkerStore.js [hunk 4]
```diff
@@ -241,8 +255,12 @@ RedisWorkerStore.prototype.logCommand = function (domain, command) {
       command = _.flattenDeep(command).join(' ');
     }
 
-    this.commandLog.push(domain + ': ' + command);
+    const entry = { command: `${domain}: ${command}` };
```
Comment (gotrevor) [inline lib/stores/redisWorkerStore.js]: why not just:
`const entry = { domain, command };`

File: lib/stores/redisWorkerStore.js (hunk 1 above)
Comment (jaquadro) [inline lib/stores/redisWorkerStore.js]: Is tap a universal promise construct?  I'm a bit more cagey when promise libraries are getting mixed.

File: lib/stores/redisWorkerStore.js (hunk 2 above)
Comment (jaquadro) [inline lib/stores/redisWorkerStore.js]: It's odd, but I'm doing it this way to definitively preserve the order of commands in the log.  That wouldn't necessarily hold if I waited until I also had the result to log.

File: lib/stores/redisWorkerStore.js (hunk 1 above)
Comment (gotrevor) [inline lib/stores/redisWorkerStore.js]: Well, Q and Bluebird both have `tap`.  Are we ever going to get a promise that isn't Q or Bluebird?

Comment (jaquadro) [inline lib/stores/redisWorkerStore.js]: Not relevant to this case, but libraries that supply their own promises?  What does AWS-SDK use natively now?  Will anyone adopt native promises, which are quite skeletal?

Comment (gotrevor) [inline lib/stores/redisWorkerStore.js]: Yes, native promises do not provide `tap`, as best I can tell.  :(

However, aws-sdk lets you specify what type of promise you get back.  It seems like a pretty standard pattern, amongst libraries that construct promises (redis does something similar, eh?):
```
[~/Code] $ ag setProm
dev-tools/bin/aws-elasticbeanstalk-version-label.js
9:AWS.config.setPromisesDependency(Q.Promise);

dev-tools/bin/aws-elasticbeanstalk-versions.js
9:AWS.config.setPromisesDependency(Q.Promise);

dev-tools/bin/aws-lambda-environment-variables.js
14:AWS.config.setPromisesDependency(Q.Promise);
```

That's the only place where we do set the promises dependency, best I can tell.

Still, I think using `tap` here would be fine.

**Pull request merged**